sources/engine/Stride.Rendering/Rendering/Images/ImageEffect.cs

```diff
@@ -143,12 +148,79 @@ public void SetDepthOutput(Texture depthStencilView, params Texture[] renderTarg
         }
 
         /// <summary>
-        /// Sets the viewport to use .
+        /// Sets the viewport to use.
+        /// </summary>
+        /// <value>The viewport.</value>
+        public void SetViewport(Viewport? value)
+        {
+            if (!value.HasValue)
+            {
+                boundViewportCount = 0;
+            }
+            else
+            {
+                boundViewportCount = 1;
+                viewports[0] = value.Value;
+            }
+        }
+
+        /// <summary>
+        /// Sets the viewport to use.
         /// </summary>
-        /// <param name="viewport">The viewport.</param>
-        public void SetViewport(Viewport? viewport)
+        /// <param name="values">The viewports.</param>
+        public void SetViewports(Viewport[] values)
         {
-            this.viewport = viewport; // TODO: support multiple viewport?
+            SetViewports(values.Length, values);
+        }
+
+        /// <summary>
+        /// Sets the viewport to use.
+        /// </summary>
+        /// <param name="viewportCount">The viewports.</param>
+        /// <param name="values">The viewports.</param>
+        public void SetViewports(int viewportCount, Viewport[] values)
```

Comment:
It is public in the command list and there was a todo comment about this. I think this is mostly done for convenience.
Like you can hold your own viewport buffer with 12 elements but only 2 of them are active.
This method would allow you to avoid unneccesary array allocations.
Could also use a Span<ViewPort> as an alternative but this is true for most of the methods in Stride so I though I will avoid this here to not mess up the code style.

Reply:
At least update the comment to describe that this is what this method does - takes first count elements from the array.